ssl/extensions.cc

```diff
@@ -3089,6 +3089,46 @@ bool ssl_negotiate_alps(SSL_HANDSHAKE *hs, uint8_t *out_alert,
   return true;
 }
 
+// Adds pha_ext to the client hello
+static bool ext_pha_add_clienthello(const SSL_HANDSHAKE *hs, CBB *out,
+                                    CBB *out_compressible, ssl_client_hello_type_t type)  {
+  const SSL *const ssl = hs->ssl;
+
+  // post_handshake_auth is not supported in TLS versions < 1.3. Also checking
+  // if extension has been enabled by client.
+  if(hs->min_version < TLS1_3_VERSION || ssl->s3->pha_enabled != 1) {
```

Comment:
min_version indicates the minimum accepted and max_version indicates maximum accepted version. It could be the case that min_version is not 1.3 so you're correct, we should check against max_version. But I'm not sure if that's enough to ensure that 1.3 is being negotiated. Something to keep in mind for testing.